src/lib/HST/CoreAlgorithm.hs

```diff
@@ -137,24 +107,22 @@ createRekMatch
   -> S.Exp a   -- ^ The error expression for pattern-matching failures.
   -> [[Eqs a]] -- ^ Groups of equations (see 'groupByFirstPatType').
   -> Sem r (S.Exp a)
-createRekMatch vars er =
-  foldr (\eqs mrhs -> mrhs >>= match vars eqs) (return er)
+createRekMatch vars er = foldr (\eqs mrhs -> mrhs >>= match vars eqs)
+  (return er)
 
 -- | Creates a case expression that performs pattern matching on the variable
 --   bound by the given variable pattern.
-makeRhs
-  :: (Members '[Env a, Fresh, GetOpt, Report] r, S.EqAST a)
-  => S.Pat a   -- ^ The fresh variable pattern to match.
-  -> [S.Pat a] -- ^ The remaing fresh variable patterns.
-  -> [Eqs a]   -- ^ The equations.
-  -> S.Exp a   -- ^ The error expression for pattern-matching failures.
-  -> Sem r (S.Exp a)
+makeRhs :: (Members '[Env a, Fresh, GetOpt, Report] r, S.EqAST a)
+        => S.Pat a   -- ^ The fresh variable pattern to match.
+        -> [S.Pat a] -- ^ The remaing fresh variable patterns.
+        -> [Eqs a]   -- ^ The equations.
+        -> S.Exp a   -- ^ The error expression for pattern-matching failures.
+        -> Sem r (S.Exp a)
 makeRhs x xs eqs er = do
   alts <- computeAlts x xs eqs er
   return (S.Case S.NoSrcSpan (S.patToExp x) alts)
 
 -- TODO remove redundand types
```

Comment:
In Floskell it is important to place comments before the code they refer to (except for `-- ^` comments). For example, if the comment belongs to the `return`, you would have to write

```haskell
makeRhs x xs eqs er = do
  alts <- computeAlts x xs eqs er
  -- TODO remove redundand types
  return (S.Case S.NoSrcSpan (S.patToExp x) alts)
```

However, in the original code the comment was placed between

Reply:
I think it can be removed. Past me should have really add a more descriptive TODO.  

I think the TODO meant that at some point i always generated all constructor patterns with new variable patterns and didn't check if the data type was partially matched before transformation.

Reply:
Thanks : )

```suggestion
```